src/lib/ptree/json.lua

```diff
@@ -86,6 +103,50 @@ local function read_json_string(input)
    return table.concat(parts)
 end
 
+local function read_json_array(input)
+   skip_whitespace(input)
+   consume(input, "[")
+   skip_whitespace(input)
+   local ret = {}
+   if not check(input, "]") then
+      repeat
+         skip_whitespace(input)
+         local v = read_json_type(input)
+         skip_whitespace(input)
+         table.insert(ret, v)
+      until not check(input, ",")
+      skip_whitespace(input)
+      consume(input, "]")
+   end
+   skip_whitespace(input)
+   return ret
+end
+
+local function read_json_scalar(input)
+   skip_whitespace(input)
+   local v = take_while(input, '[0-9%-%.truefalsn]')
```

Comment:
This is still limited in that it is a subset of number syntax (e.g. no +, no E100, etc); that's OK but we should document with a comment.

Reply:
I've added `+` and `E` to this (alongside `-` and `e` which already existed in the `take_while`) - the syntax checking here is a little loose as it matches by-character but I *think* this means it supports the full spec of JSON representable numbers, as I think these can all be parsed successfully by Lua's `tonumber()` anyway.